### PR TITLE
[Upstream PWA] Cache POST response for react components

### LINKED ISSUE #302: [Upstream Demo] Libraries in React are not being cached
Like Directory, Admin Settings, Sort Popup etc.

### PR DESCRIPTION
**Pull request by shubhsherl**
Closes #302 

Add indexedDb to manage POST requests for react related modules. First a network request will be made, and response will be send to the page, and to the indexedDb. If the network request fails, then try to get from Db(if available).

### CODE REVIEW

**Comment by ear-dev** [comment]
@Shailesh351 can you review please?

**Comment by ear-dev** [comment]
@Shailesh351 have you had a chance yet to look at this one?  Thanks......

File: public/sw.js
Comment (Shailesh351) [inline public/sw.js]: @shubhsherl Instead of every-time opening database here. Can we do like only open when db is null? Will improve performance a bit.

Rest of the things looks perfect. Great work!! 👏 🎉

**Comment by ear-dev** [comment]
@Shailesh351 looks like @shubhsherl made the fix?  Go ahead and approve if you think it's ready.....  I will merge it...  Thanks guys!

File: public/sw.js
Comment (ear-dev) [reply public/sw.js]: @shubhsherl did you also port this to the upstream PRs?  Just checking..........

Comment (shubhsherl) [reply public/sw.js]: @ear-dev I'll update in upstream PRs after this one gets merged.

**Review by Shailesh351 — APPROVED** [review]
LGTM 🎉  Thanks @shubhsherl 
I found one bug. I will create issue for the same. In the offline mode, user is not able to toggle sort/group by options.

**Comment by ear-dev** [comment]
Thanks guys!

**Pull request merged**
